### PR TITLE
8310143: RandomCommandsTest fails due to unexpected VM exit code after JDK-8282797

### PR DESCRIPTION
**Pull request by TobiHartmann**
The fix for [JDK-8282797](https://bugs.openjdk.org/browse/JDK-8282797) missed that randomly generated Compile Commands can be invalid. I verified with the corresponding seeds that all occurrences of this intermittent failure are now fixed.

Thanks,
Tobias

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8310143](https://bugs.openjdk.org/browse/JDK-8310143): RandomCommandsTest fails due to unexpected VM exit code after JDK-8282797 (**Bug** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [61f44405](https://git.openjdk.org/jdk/pull/14514/files/61f4440523434526967bdf1642ee946f289d0479)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14514/head:pull/14514` \
`$ git checkout pull/14514`

Update a local copy of the PR: \
`$ git checkout pull/14514` \
`$ git pull https://git.openjdk.org/jdk.git pull/14514/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14514`

View PR using the GUI difftool: \
`$ git pr show -t 14514`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14514.diff">https://git.openjdk.org/jdk/pull/14514.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14514#issuecomment-1594473151)

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good.

File: test/hotspot/jtreg/compiler/compilercontrol/share/MultiCommand.java [hunk 1]
```diff
@@ -78,12 +80,13 @@ public static AbstractTestBase generateRandomTest(boolean validOnly) {
                 md = AbstractTestBase.getValidMethodDescriptor(exec);
             } else {
                 md = AbstractTestBase.METHOD_GEN.generateRandomDescriptor(exec);
+                isValid = false;
```
Comment (robcasloz) [inline test/hotspot/jtreg/compiler/compilercontrol/share/MultiCommand.java:83]: Is there a (remote) chance that `generateRandomDescriptor()` generates a valid descriptor? Would the compiler control test fail in that case due to a "false negative" ("expected to fail but did not fail")? Maybe a clarifying comment here would help.

Comment (TobiHartmann) [reply test/hotspot/jtreg/compiler/compilercontrol/share/MultiCommand.java]: Yes, I think that could happen. I updated that line to use `md.isValid()`.

**Comment by TobiHartmann** [comment]
Thanks for the reviews, Vladimir and Roberto! My testing caught failures because I accidentally declared the `isValid` field outside of the loop but it should be re-set to `true` after each iteration.

**Comment by TobiHartmann** [comment]
I screwed up the merge and re-opened the PR with https://github.com/openjdk/jdk/pull/14538. Sorry for the noise.

**Pull request closed without merging**